sigstore/sign.py

```diff
@@ -82,74 +89,106 @@ class Signer:
     The primary API for signing operations.
     """
 
-    def __init__(self, *, fulcio: FulcioClient, rekor: RekorClient):
+    def __init__(self, identity_token: str, cache: bool = True) -> None:
         """
         Create a new `Signer`.
 
-        `fulcio` is a `FulcioClient` capable of connecting to a Fulcio instance
-        and returning signing certificates.
+        `identity_token` is the identity token used to request a signing certificate
+        from Fulcio.
 
-        `rekor` is a `RekorClient` capable of connecting to a Rekor instance
-        and creating transparency log entries.
+        `cache` determines whether the signing certificate and ephemeral private key
+        should be reused (until the certificate expires) to sign different artifacts.
+        Default is `True`.
         """
-        self._fulcio = fulcio
-        self._rekor = rekor
+        self._identity_token: str = identity_token
+        self.cache: bool = cache
```

Comment:
One thought: rather than retaining `cache` as an attribute, we could do:

```python
self.__cached_private_key = None
if cache:
    self._cached_private_key = keygen(...)
```

This eliminates the dual state checks below (we can just check `_cached_private_key`).